ledfx/virtuals.py

```diff
@@ -448,19 +438,18 @@ def set_effect(self, effect, fallback: Optional[float] = None):
             if fallback is not None:
                 _LOGGER.info("Fallback requested")
                 if self._active_effect is not None:
-                    if self.fallback_effect_type is None:
+                    if not self.fallback_active:
```

Comment:
This is tricky - can you use few guard clauses in here to try and simplify this triple nest?

Reply:
None of it is "do nothing and return" though, its in line with set effect. The three paths have different functionality

I can muse on refactoring, but I am not sure there is a magic bullet here.

Reply:
I have mashed it down as much as I dare, and demoted some debug to just a comment which saves a clause.

I will need to have test some deeper corners that are outside of what I have done in pytest.